lib/ansible/modules/cloud/google/gcp_bigquery_dataset.py

```diff
@@ -420,7 +507,7 @@ def fetch_resource(module, link, kind, allow_not_found=True):
 
 
 def self_link(module):
-    return "https://www.googleapis.com/bigquery/v2/projects/{project}/datasets/{name}".format(**module.params)
+    return "https://www.googleapis.com/bigquery/v2/projects/{project}/datasets/{dataset_id}".format(**module.params)
```

Comment:
This will break the module. Can we use the "{{name}}" parameter for the self_link?

If not, we can override it.